staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -838,11 +838,13 @@ func (s *GenericAPIServer) InstallAPIGroups(apiGroupInfos ...*APIGroupInfo) erro
 	for _, apiGroupInfo := range apiGroupInfos {
 		// Do not register empty group or empty version.  Doing so claims /apis/ for the wrong entity to be returned.
 		// Catching these here places the error  much closer to its origin
```

Comment:
prefer a length check and early error:

```go
if len(apiGroupInfo.PrioritizedVersions) == 0 {
  return fmt.Errorf("no version priority set for %#v", *apiGroupInfo)
}
```

Reply:
ok, this is good. And with this validation, we may not worry about the later codes logic.